src/ng/parse.js

```diff
@@ -38,20 +38,30 @@ var $parseMinErr = minErr('$parse');
 
 
 function ensureSafeMemberName(name, fullExpression) {
+  if (name === "__defineGetter__" || name === "__defineSetter__"
+      || name === "__lookupGetter__" || name === "__lookupSetter__"
+      || name === "__proto__") {
+    throw $parseMinErr('isecfld',
+        'Attempting to access a disallowed field in Angular expressions! '
+        + 'Expression: {0}', fullExpression);
+  }
+  return name;
+}
```

Comment:
ensureSafeMemberName is called in three cases.
* retrieving an identifier. Eg `foo`
* retrieving a non-computed member. Eg `foo.bar`
* retrieving a computed member. Eg `foo[bar]`

getStringValue is only used in the later. This is the expected behavior